pkg/api/agentPoolOnlyApi/v20180331/validate.go

```diff
@@ -141,108 +141,121 @@ func validateVNET(a *Properties) error {
 
 	n := a.NetworkProfile
 
-	if n == nil {
-		return ErrorNilNetworkProfile
-	}
+	// validate network profile settings
+	if n != nil {
+		if string(n.NetworkPlugin) == string(Azure) || string(n.NetworkPlugin) == string(Kubenet) {
```

Comment:
is it easier to read using `switch` and `case`?

Reply:
In this case, Azure and Kubenet is handled through the same code path, so switch does not help.

NetworkPlugin can be instantiated by taking a string, I think it can not be compared directly to constant Azure or Kubenet, even though the string value are the same.

The model now requires case matching too.

Reply:
In `pkg/acsengine/testdata/agentPoolOnly/v20180331/agentsWithFullNetworkProfile.json`, it says `azure`, though..

Reply:
Actually tested (NetworkPlugin("azure") == Azure) is true, will change the code to make it simple.

"azure" is the serialized representation of object Azure, NetworkPlugin("Azure") will not equal to object Azure. So it is case sensitive.